server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -1930,18 +1931,13 @@ private boolean upgradeRunningVirtualMachine(Long vmId, Long newServiceOfferingI
     public HashMap<String, VolumeStatsEntry> getVolumeStatistics(long clusterId, String poolUuid, StoragePoolType poolType, List<String> volumeLocators, int timeout) {
         List<HostVO> neighbors = _resourceMgr.listHostsInClusterByStatus(clusterId, Status.Up);
         StoragePoolVO storagePool = _storagePoolDao.findPoolByUUID(poolUuid);
-        for (HostVO neighbor : neighbors) {
-            // apply filters:
-            // - managed storage
-            // - local storage
-            if (storagePool.isManaged() || storagePool.isLocal()) {
```

Comment:
@slavkap what is the impact on volume stats for managed / local storage pool ? can this cause regression?

Reply:
The logic was correct for managed/local storages, but for unmanaged I've descried in the description what is happening. I have tested with NFS unmanaged storage and StorPool as well, and it should get the volumes only on the current host

Reply:
@slavkap For non-managed & shared pools, the `volumeLocators` passed to the method is directly used earlier, instead from the host(s). Now, `volumeLocators` in the method arg, is not used anymore. You can remove the arg if not required. Please check and update accordingly.

Reply:
removed volumeLocators